cli/assets/templates/zig/build.zig

```diff
@@ -1,42 +1,46 @@
 const std = @import("std");
-const expect = std.testing.expect;
-const expectEqualStrings = std.testing.expectEqualStrings;
 
-/// Panics if current zig version is below the minimal required.
-fn check_zig_version() !void {
-    const zig_version = @import("builtin").zig_version;
-    if (zig_version.major > 0) return;
-    if (zig_version.minor > 10) return;
+// Returns true if the version includes https://github.com/ziglang/zig/pull/10572/commits.
+// When this is false, trying to place the stack first will result in data corruption.
+fn version_supports_stack_first(zig_version: std.SemanticVersion) !bool {
+    // TODO, also allow corresponding 0.9.x+ if and when the fix has been backported.
+    // As of today it hasn't been backported yet: https://github.com/ziglang/zig/commits/0.9.x
+    return zig_version.order(try std.SemanticVersion.parse("0.10.0-dev.258")).compare(.gte);
```

Comment:
```suggestion
    if (zig_version.pre) |pre| {
        // assert we understand the format
        try std.testing.expect(pre.len >= 5);
        try expectEqualStrings(pre[0..4], "dev.");
        try std.fmt.parseUnsigned(u32, pre[4..], 10);
        
        return zig_version.order(try std.SemanticVersion.parse("0.10.0-dev.258")).compare(.gte);
    } else return zig_version.order(try std.SemanticVersion.parse("0.9.1")).compare(.gte);
```

Reply:
We shouldn't need to parse the `pre` tag. Semantic versioning already specifies an ordering over `pre` tags, which is implemented in the `SemanticVersion` class.

Reply:
See 11.4 https://semver.org/#spec-item-11

Reply:
Yes, however i added assertion just to make sure there's nothing like `0.10.0-bootstrap.1`. Something like `0.10.0-alpha.1` would be less than `0.10.0-dev.258` by these rules, but that would be wrong, so i think we better handle only nighlty's `dev-*` and stable.

Reply:
IMO them doing that would be a violation of the semver spec, but I don't know their processes well enough to be confident that they won't do it anyway. Went ahead and put in a check for "dev.".